### DISCUSSION TITLE
[ZK Stack] - how to  modify the gas fee?

### DISCUSSION THREAD

**Discussion by SuccinctPaul** [zkSync-Community-Hub/zksync-developers#670] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [X] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.78.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

* For now, the gas fee is calculated as same as zksync-era. What should we do if we want to modify the gas fee?
* Can we make it lower or higher only modifying the config variables in `chain.toml`?

### Expected Behavior

* modify the gas fee

### Current Behavior

No idea.

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
To modify the gas fees in the ZKsync Era, particularly for components like the zkEVM and the fee withdrawer, you cannot directly adjust them through the `chain.toml` configuration file alone. The gas fee structure in ZKsync Era is influenced by several factors, including L1 gas prices and the specific computational and storage requirements of transactions.

Here are some steps and considerations for adjusting gas fees:

1. **Understand the Fee Model**: Familiarize yourself with how gas fees are calculated in ZKsync Era. This includes understanding the impact of L1 gas prices and the cost of zero-knowledge proof generation. More details can be found in the [ZKsync Fee Model documentation](https://docs.zksync.io/build/developer-reference/fee-model).

2. **Modify Smart Contract Logic**: If feasible, consider optimizing the smart contract logic to reduce the computational resources required, thereby potentially lowering the gas fees.

3. **Operator Settings**: Some adjustments can be made by the operator in terms of how fees are calculated and applied. This requires access to the system's backend configuration, which may not be directly exposed via simple configuration files like `chain.toml`.

4. **Consult Documentation and Community**: For specific guidance on modifying operational parameters within ZKsync Era's infrastructure, consult the official documentation or seek advice from the ZKsync developer community.

Unfortunately, there is no straightforward way to adjust gas fees merely by changing configuration variables in `chain.toml`. It typically involves a more comprehensive approach including smart contract optimization and potentially modifying the operator's fee policies.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply]
@SuccinctPaul did the suggestion above help? If not could you downvote it and let us know what else you've tried.

**Reply by SuccinctPaul** [reply]
> @SuccinctPaul did the suggestion above help? If not could you downvote it and let us know what else you've tried.

@bxpana Hi, it doesn't help, now still struggle with this problem.

**Reply by bxpana** [reply, marked as answer]
Things in the `chain.toml` and other configs allow you to control different aspects of the price. Here's a doc with a bunch of details of how things can be configured https://github.com/matter-labs/zksync-era/blob/main/docs/guides/advanced/07_fee_model.md
